Makefile

```diff
@@ -60,17 +64,53 @@ PROJECT_CMODULES   += crc32 compress deflate trees
 
 PROJECT_OCAML_LIBS = unix str #OCaml changed libstr into libcamlstr and "you are not supposed to link with -lstr"
 PROJECT_LIBS       = unix camlstr
+
+PROJECT_RESOURCES = weidu_resources
+
 .PHONY: weidu
-weidu:  $(PROJECT_EXECUTABLE)
+weidu: $(PROJECT_EXECUTABLE)
+
 $(PROJECT_EXECUTABLE) : $(PROJECT_MODULES:%=$(OBJDIR)/%.$(CMO)) \
                         $(PROJECT_CMODULES:%=$(OBJDIR)/%.$(OBJEXT))
 	@$(NARRATIVE) Linking $(COMPILETOWHAT) $@
+
+ifeq ( $(shell uname -s),"Linux")
 	$(CAMLLINK) -o $@ \
-                    $(PROJECT_OCAML_LIBS:%=%.$(CMXA)) \
-                    $(PROJECT_LIBS:%=-cclib -l%) \
-                    $(PROJECT_CLIBS:%=-cclib %) \
-                    $^
-	cp $(PROJECT_EXECUTABLE) .
+		$(PROJECT_OCAML_LIBS:%=%.$(CMXA)) \
+		$(PROJECT_LIBS:%=-cclib -l%) \
+		$(PROJECT_CLIBS:%=-cclib %) \
+		$^
+		cp $(PROJECT_EXECUTABLE) .
+endif
+
+ifeq ( $(shell uname -s),"Darwin")
+	$(CAMLLINK) -o $@ \
+		$(PROJECT_OCAML_LIBS:%=%.$(CMXA)) \
+		$(PROJECT_LIBS:%=-cclib -l%) \
+		$(PROJECT_CLIBS:%=-cclib %) \
+		$^
+		cp $(PROJECT_EXECUTABLE) .
+endif
+
+ifeq ($(OS),Windows_NT)
+
+	@echo VERSION_MAJOR IS $(VERSION_MAJOR)
+	@echo VERSION_MINOR IS $(VERSION_MINOR)
+
+	sed -i.backup "s/FILEVERSION\W\+[[:digit:]]\+,[[:digit:]]\+,[[:digit:]]\+,[[:digit:]]\+/FILEVERSION\t\t$(VERSION_MAJOR),$(VERSION_MINOR),0,0/gI" windows_resources/weidu_resources.rc
+	sed -i.backup "s/\"FileVersion\",\ \"[[:digit:]]\+.[[:digit:]]\+\"/\"FileVersion\",\ \"$(VERSION_MAJOR).$(VERSION_MINOR)\"/gI" windows_resources/weidu_resources.rc
+	sed -i.backup "s/\"ProductVersion\",\ \"[[:digit:]]\+.[[:digit:]]\+\"/\"ProductVersion\",\ \"$(VERSION_MAJOR).$(VERSION_MINOR)\"/gI" windows_resources/weidu_resources.rc
+
+	x86_64-w64-mingw32-windres -i windows_resources/weidu_resources.rc -o $(OBJDIR)/weidu_resources.o
```

Comment:
Can you put the name of the windres executable in a variable in the Configuration file?

Reply:
Also done.